dao/src/main/java/org/thingsboard/server/dao/sql/job/JobRepository.java

```diff
@@ -45,13 +45,19 @@ Page<JobEntity> findByTenantIdAndSearchText(@Param("tenantId") UUID tenantId,
     @Query("SELECT j FROM JobEntity j WHERE j.id = :id")
     JobEntity findByIdForUpdate(UUID id);
 
-    boolean existsByKeyAndStatusIn(String key, List<JobStatus> statuses);
+    @Query("SELECT j FROM JobEntity j WHERE j.tenantId = :tenantId AND j.key = :key " +
+            "ORDER BY j.createdTime DESC")
+    JobEntity findLatestByTenantIdAndKey(@Param("tenantId") UUID tenantId, @Param("key") String key);
+
+    boolean existsByTenantIdAndKeyAndStatusIn(UUID tenantId, String key, List<JobStatus> statuses);
 
     boolean existsByTenantIdAndTypeAndStatusIn(UUID tenantId, JobType type, List<JobStatus> statuses);
 
     @Lock(LockModeType.PESSIMISTIC_WRITE) // SELECT FOR UPDATE
     @Query("SELECT j FROM JobEntity j WHERE j.tenantId = :tenantId AND j.type = :type " +
-           "AND j.status = :status ORDER BY j.createdTime ASC, j.id ASC")
+            "AND j.status = :status ORDER BY j.createdTime ASC, j.id ASC")
     JobEntity findOldestByTenantIdAndTypeAndStatusForUpdate(UUID tenantId, JobType type, JobStatus status, Limit limit);
 
+    void deleteByTenantId(UUID tenantId);
```

Comment:
Always use `@Query` for bulk-deletion methods, because by default Hibernate does "SELECT ALL" and deletes entities one by one